WireDomain/Sources/WireDomain/Extensions/CoreData/NSManagedObjectContext.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+
+extension NSManagedObjectContext {
+
+    /// Saves or rollback changes if saving fails
+    ///
+    /// This is a lighter implementation of the `saveOrRollback`objc- method for `WireDataModel`.
+    func saveOrRevert() throws {
```

Comment:
across codebase, we often don't explicitly save context, I figured I'd create a method in WireDomain (for now) to save context or rollback changes, a similar but much lighter version of the legacy `saveOrRollback` method from WireDataModel.

Reply:
This is fine for now, but as I mentioned in another comment, I believe we need some tight control about when we save.